packages/headless/src/features/insight-search/insight-search-state.ts

```diff
@@ -0,0 +1,52 @@
+import {InsightAPIErrorStatusResponse} from '../../api/service/insight/insight-api-client';
+import {
+  InsightQueryResponse,
+  SearchResult,
+} from '../../api/service/insight/query/query-response';
+
+export interface InsightSearchState {
+  isLoading: boolean;
+  error?: InsightAPIErrorStatusResponse | null;
```

Comment:
I don't think we should have `| null` since `error` is already marked as optional.